components-java-sdk/src/main/java/io/dapr/components/domain/pubsub/exceptions/InvalidAcknowledgementMessageException.java

```diff
@@ -11,22 +11,14 @@
  * limitations under the License.
  */
 
-package io.dapr.components.domain.pubsub;
+package io.dapr.components.domain.pubsub.exceptions;
 
-import java.util.Optional;
+import io.dapr.components.domain.pubsub.PullMessageAcknowledgement;
 
-public record PullMessagesRequest(Topic topic, String ackMessageId, Optional<String> ackErrorMessage) {
-
-  /**
-   * Conversion constructor.
-   *
-   * @param other The Protocol Buffer representation of a PullMessagesRequest.
-   */
-  public PullMessagesRequest(dapr.proto.components.v1.Pubsub.PullMessagesRequest other) {
-    this(new Topic(other.getTopic()),
-        other.getAckMessageId(),
-        other.hasAckError()
-            ? Optional.of(other.getAckError().getMessage())
-            : Optional.empty());
-  }
+/**
+ * Signals that an error was found while trying to parse a
+ * {@link dapr.proto.components.v1.Pubsub.PullMessagesRequest} into
+ * a {@link PullMessageAcknowledgement} because it came with a topic field set.
+ */
+public class InvalidAcknowledgementMessageException extends IllegalArgumentException {
```

Comment:
Consider marking these exception types as "internal"?  It doesn't seem like end users would ever handle them, would they?

Reply:
True. I wrongly assumed they could handle them but your are right: they would never be bubbled up through user code.


Java  doesn't have an `internal` visibility as C#. The closest is "package/default" visibility and, in this case, package  is `io.dapr.components.domain.pubsub.exceptions`. I will just move them up in the hierarchy.

Reply:
Fixed on 8ae6ea3dadea9be5267014ae625a0877a16efeb5